scapy/layers/http.py

```diff
@@ -663,19 +671,36 @@ def http_request(host, path="/", port=80, timeout=3,
     }
     http_headers.update(headers)
     req = HTTP() / HTTPRequest(**http_headers)
-    tcp_client = TCP_client.tcplink(HTTP, host, port, debug=verbose,
-                                    iface=iface)
     ans = None
-    if iptables:
-        ip = tcp_client.atmt.dst
+
+    # Open a socket
+    if iface is not None:
+        raw = True
+    if raw:
+        # Use TCP_client on a raw socket
         iptables_rule = "iptables -%c INPUT -s %s -p tcp --sport 80 -j DROP"
-        assert(os.system(iptables_rule % ('A', ip)) == 0)
+        if iptables:
+            host = str(Net(host))
+            assert(os.system(iptables_rule % ('A', host)) == 0)
+        sock = TCP_client.tcplink(HTTP, host, port, debug=verbose,
+                                  iface=iface)
+    else:
+        # Use a native TCP socket
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.connect((host, port))
+        sock = StreamSocket(sock, HTTP)
+    # Send the request and wait for the answer
     try:
-        ans = tcp_client.sr1(req, timeout=timeout, verbose=verbose)
+        ans = sock.sr1(
+            req,
+            session=TCPSession(app=True),
+            timeout=timeout,
+            verbose=verbose
+        )
     finally:
-        tcp_client.close()
-        if iptables:
-            assert(os.system(iptables_rule % ('D', ip)) == 0)
+        sock.close()
+        if raw and iptables:
+            assert(os.system(iptables_rule % ('D', host)) == 0)
```

Comment:
We should call `host = str(Net(host))` again before `os.system()` to make sure that `host` in the not a shell command that can be injected into iptables_rule.